src/lib.rs

```diff
@@ -0,0 +1,538 @@
+                    break;
+                }
+                val.pop().unwrap();
+            }
+            (line_off, key, val)
+        };
+
+        let mut tests = HashMap::new();
+        let mut line_off = 0;
+        while line_off < lines.len() {
+            let indent = indent_level(line_off);
+            if indent == lines[line_off].len() {
+                line_off += 1;
+                continue;
+            }
+            let (test_name, val) = key_val(line_off, indent);
+            if !val.is_empty() {
+                panic!(
+                    "Test name '{}' can't have a value on line {}.",
+                    test_name, line_off
+                );
+            }
+            match tests.entry(test_name.to_lowercase()) {
+                Entry::Occupied(_) => panic!(
+                    "Command name '{}' is specified more than once, line {}.",
+                    test_name, line_off
+                ),
+                Entry::Vacant(e) => {
+                    line_off += 1;
+                    let mut test = Test {
+                        status: None,
+                        stderr: None,
+                        stdout: None,
+                    };
+                    while line_off < lines.len() {
+                        let sub_indent = indent_level(line_off);
+                        if sub_indent == lines[line_off].len() {
+                            line_off += 1;
+                            continue;
+                        }
+                        if sub_indent == indent {
+                            break;
+                        }
+                        let (end_line_off, key, val) = key_multiline_val(line_off, sub_indent);
+                        line_off = end_line_off;
+                        match key {
+                            "status" => {
+                                let val_str = val.join("\n");
+                                let v_lower = val_str.to_lowercase();
+                                if v_lower != "success" && v_lower != "error" {
+                                    panic!("Unknown status '{}' on line {}", val_str, line_off);
+                                }
+                                test.status = Some(v_lower);
+                            }
+                            "stderr" => {
+                                test.stderr = Some(val);
+                            }
+                            "stdout" => {
+                                test.stdout = Some(val);
+                            }
+                            _ => panic!("Unknown key '{}' on line {}.", key, line_off),
+                        }
+                    }
+                    e.insert(test);
+                }
+            }
+        }
+        tests
+    }
+
+    /// Check for the case where the user has a test such as:
+    ///   [Test]
+    /// but test_cmds doesn't have any test called "Test". This is almost certainly a bug, in the
```

Comment:
I'm not sure I understand this part. What does it mean to have a "test such as `[Test]`"

Reply:
Sorry ignore me. I get it now... I just misread the comment :man_facepalming:

Reply:
Sorry, it's also using the old syntax...